File: Data Structures & Algorithms/majority-element-ii/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        res = []
        d = {}

        m = int(len(nums) / 3) + 1
        for i in nums:
            d[i] = d.get(i, 0) + 1

            if d[i] == m:
                res.append(i)


        return res

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> x;
        
        for(int i : nums)
        {
            if(x.find(i) == x.end())
            {
                x[i] = 1;
            }
            else return true;
        }
        return false;
    }
};
